tests/components/smartthings/test_config_flow.py

```diff
@@ -224,128 +283,159 @@ async def test_cloudhook_app_created_then_show_wait_form(
 
-async def test_multiple_config_entry_created_when_installed(
-    hass, app, locations, installed_apps, smartthings_mock
-):
-    """Test a config entries are created for multiple installs."""
-    assert await async_setup_component(hass, "persistent_notification", {})
+    assert flow.location_id == location_id
+    assert result["type"] == data_entry_flow.RESULT_TYPE_EXTERNAL_STEP
+    assert result["step_id"] == "authorize"
+    assert result["url"] == format_install_url(app_id, location_id)
+
+
+async def test_step_authorize_advances(hass):
+    """Test authorize step advances when completed."""
+    installed_app_id = str(uuid4())
+    refresh_token = str(uuid4())
+    flow = SmartThingsFlowHandler()
+    flow.hass = hass
+
+    result = await flow.async_step_authorize(
+        {CONF_INSTALLED_APP_ID: installed_app_id, CONF_REFRESH_TOKEN: refresh_token}
+    )
+
+    assert flow.installed_app_id == installed_app_id
+    assert flow.refresh_token == refresh_token
+    assert result["type"] == data_entry_flow.RESULT_TYPE_EXTERNAL_STEP_DONE
+    assert result["step_id"] == "install"
+
+
+async def test_step_install_creates_entry(hass, location, smartthings_mock):
+    """Test a config entry is created once the app is installed."""
     flow = SmartThingsFlowHandler()
     flow.hass = hass
-    flow.access_token = str(uuid4())
-    flow.app_id = app.app_id
     flow.api = smartthings_mock
+    flow.access_token = str(uuid4())
+    flow.app_id = str(uuid4())
+    flow.installed_app_id = str(uuid4())
+    flow.location_id = location.location_id
     flow.oauth_client_id = str(uuid4())
     flow.oauth_client_secret = str(uuid4())
-    for installed_app in installed_apps:
-        data = {
-            CONF_REFRESH_TOKEN: str(uuid4()),
-            CONF_LOCATION_ID: installed_app.location_id,
-            CONF_INSTALLED_APP_ID: installed_app.installed_app_id,
-        }
-        hass.data[DOMAIN][CONF_INSTALLED_APPS].append(data)
-    install_data = hass.data[DOMAIN][CONF_INSTALLED_APPS].copy()
+    flow.refresh_token = str(uuid4())
 
-    result = await flow.async_step_wait_install({})
-
-    assert not hass.data[DOMAIN][CONF_INSTALLED_APPS]
+    result = await flow.async_step_install()
 
     assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
-    assert result["data"]["app_id"] == installed_apps[0].app_id
-    assert result["data"]["installed_app_id"] == installed_apps[0].installed_app_id
-    assert result["data"]["location_id"] == installed_apps[0].location_id
+    assert result["data"]["app_id"] == flow.app_id
+    assert result["data"]["installed_app_id"] == flow.installed_app_id
+    assert result["data"]["location_id"] == flow.location_id
     assert result["data"]["access_token"] == flow.access_token
-    assert result["data"]["refresh_token"] == install_data[0][CONF_REFRESH_TOKEN]
+    assert result["data"]["refresh_token"] == flow.refresh_token
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5359319)